src/components/simulation/FooterButtons.vue

```diff
@@ -1,20 +1,43 @@
 <script lang="ts" setup>
 import { useSimulationStore } from '@/stores/simulations'
 import { useRouter } from 'vue-router'
-
 const router = useRouter()
 const simulationStore = useSimulationStore()
-
 function clickButtonCancel() {
   router.push('/roof-selection')
 }
-
 function clickButtonPrevious() {
-  simulationStore.goToPreviousStep()
+  if (simulationStore.currentStep == 3) {
```

Comment:
I think we should put this navigation in the simulation store, and the button should only call `goToPreviousStep()` or `gotToNextStep()`